### PR TITLE
[Issue 2] Support reading properties from the standard uPortal locati…

### PR DESCRIPTION
**Pull request by jhelmer-unicon**
##### Checklist

-   [x] the [individual contributor license agreement][] is signed
-   [x] commit message follows [commit guidelines][]

##### Description of change
Update to read property files from the standard uPortal locations.

[individual contributor license agreement]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#individual-contributor-license-agreement
[commit guidelines]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#commit
[message properties]: https://github.com/Jasig/uPortal/tree/master/uportal-war/src/main/resources/properties/i18n
[WCAG 2.0 AA]: https://www.w3.org/WAI/WCAG20/quickref/?levels=aaa&technologies=smil%2Cpdf%2Cflash%2Csl

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -50,6 +50,9 @@ dependencies {
     compile("org.jasig.portal:uPortal-soffit-renderer:${soffitVersion}")
     compile('org.springframework.boot:spring-boot-starter-cache')
     compile('org.springframework.boot:spring-boot-starter-web')
+    compile("org.jasig.portal:uPortal-spring:${uportalLibsVersion}")
```
Comment (drewwills) [inline build.gradle]: I'm concerned this element will bring in half of uPortal as transitive dependencies.  Can we `exclude` the same things we do in Maven?

File: gradle.properties [hunk 2]
```diff
@@ -16,3 +16,6 @@ ehcacheVersion=2.10.2
 httpclientVersion=4.5.2
 jstlVersion=1.2
 soffitVersion=5.0.0-M4
+uportalLibsVersion=5.0.0-M4
```
Comment (drewwills) [inline gradle.properties]: `uportalLibsVersion` and `soffitVersion` basiclly specify the same thing.  We should probably define this property once. 
 Your name is more general and better suited.

File: src/main/java/org/apereo/portal/soffits/SoffitSamplesApplication.java [hunk 3]
```diff
@@ -19,19 +19,34 @@
...
 @SoffitApplication
+@EnableAutoConfiguration(exclude = {
+        DataSourceAutoConfiguration.class,
+        HibernateJpaAutoConfiguration.class
+})
 @EnableCaching
+@EnableEncryptableProperties
+@PropertySources({
+        @PropertySource(value = "file://${portal.home}/global.properties", ignoreResourceNotFound = true),
+        @PropertySource(value = "file://${portal.home}/soffit-samples.properties", ignoreResourceNotFound = true)
```
Comment (drewwills) [inline src/main/java/org/apereo/portal/soffits/SoffitSamplesApplication.java]: What happens when `portal.home` is not defined and/or these files don't exist within it?  (We're supposed to support that.)

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -44,3 +44,6 @@ catalina.home=build
 # (See:  http://instagram.pixelunion.net/)
 #
 #instagram.api.access_token=
+
+jasypt.encryptor.password=${UP_JASYPT_KEY}
+portal.home=${PORTAL_HOME:${catalina.home}/portal}
```
Comment (drewwills) [inline src/main/resources/application.properties]: Maybe `${catalina.base}/portal}`?

File: src/main/java/org/apereo/portal/soffits/SoffitSamplesApplication.java (hunk 3 above)
Comment (ChristianMurphy) [reply src/main/java/org/apereo/portal/soffits/SoffitSamplesApplication.java]: It looks like `${portal.home}` has a fallback

https://github.com/jhelmer-unicon/soffit-samples/blob/bde9c4215f75de8de69fca8c30ff5cc337c9c476/src/main/resources/application.properties#L49

File: build.gradle (hunk 1 above)
Comment (jhelmer-unicon) [reply build.gradle]: Ah.  Yes.  I forgot to do that.

Comment (jhelmer-unicon) [reply build.gradle]: Heh.  Actually...  I don't need this at all.

File: src/main/resources/application.properties (hunk 4 above)
Comment (jhelmer-unicon) [reply src/main/resources/application.properties]: Yeah, I just tried that and it seems to work too.   Will update.

Comment (drewwills) [reply src/main/resources/application.properties]: That's awesome -- thanks!

**Comment by drewwills** [comment]
I will build a `-SNAPSHOT` based on this PR and run it in my UCR deployment that depends on values for soffit encryption keys defined in `global.properties`.

**Comment by ChristianMurphy** [comment]
Any additional changes requested? :eyes: Or is this ready to merge :white_check_mark: @apetro @drewwills

**Comment by drewwills** [comment]
This is funny -- I thought this feature was already added.  Let me have a look.

**Comment by drewwills** [comment]
LGTM :+1:

**Pull request merged**